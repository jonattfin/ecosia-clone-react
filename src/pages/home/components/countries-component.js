
import * as Images from './images';

export default function Component({ language, styles }) {
  return (
    <section className={styles['countries-section']}>
      <div className={styles['above-title']}>
        {language['where-do-you-plant-trees']}
      </div>
      <div className={styles.title}>
        {language['we-plant-30-countries']}
      </div>
      <div className={styles['countries-grid']}>
        <div className={styles['grid-item']}>
          <div><img className={styles.country} src={Images.BrazilImage} alt="brazil" /></div>
          <div className={styles['grid-item__title']}>Brazil</div>
          <div className={styles['grid-item__subtitle']}>
            {language['we-plant-in-brazil']}
          </div>
        </div>
        <div className={styles['grid-item']}>
          <img className={styles.country} src={Images.BurkinaFasoImage} alt="burkina faso" />
          <div className={styles['grid-item__title']}>Burkina Faso</div>
          <div className={styles['grid-item__subtitle']}>{language['we-plant-in-burkina-faso']}</div>
        </div>
        <div className={styles['grid-item']}>
          <img className={styles.country} src={Images.IndonesiaImage} alt="indonesia" />
          <div className={styles['grid-item__title']}>Indonesia</div>
          <div className={styles['grid-item__subtitle']}>{language['we-plant-in-indonesia']}</div>
        </div>
      </div>
    </section >
  )
}
