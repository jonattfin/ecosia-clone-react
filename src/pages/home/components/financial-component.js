import { Link } from 'react-router-dom';
import { Grid } from '@mui/material';
import styled from '@emotion/styled'

import PieComponent from './pie';
import { AboveTitle, Subtitle } from './styled-components';

const PieContainer = styled.div`
  height: 20vh;
  width: 20vw;
`;

export default function Component({ language }) {
  return (
    <Grid container spacing={2}>
      <Grid item xs={12} xl={6}>
        <AboveTitle>Monthly financial reports </AboveTitle>
        <Subtitle>Our revenue in January 2022 </Subtitle>
        <div>Our monthly reports show how much ad revenue we made from your searches, how we spent it, and how many trees this helped us plant.</div>
      </Grid>
      <Grid item xs={12} xl={6}>
        <PieContainer>
          <PieComponent />
        </PieContainer>
      </Grid>
      <Grid item xs={12} xl={12}>
        <Link to="#">Explore our financial reports...</Link>
      </Grid>
    </Grid>
  )
}
