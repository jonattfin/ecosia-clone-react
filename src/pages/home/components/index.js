import PieComponent from './responsive-pie';

const Components = {
  PieComponent
}

export default Components;
