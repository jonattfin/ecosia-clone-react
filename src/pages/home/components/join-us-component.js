import { Button } from '@mui/material';
import styled from '@emotion/styled';

import { Subtitle } from './styled-components';

const DivContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 20vh;
  background-color: aliceblue;
`;

export default function Component({ language }) {
  return (
    <DivContainer>
      <Subtitle>Join 15 million people who already use Ecosia</Subtitle>
      <div>&nbsp;</div>
      <div>
        <Button color='primary' variant="contained">Share Ecosia</Button>
      </div>
    </DivContainer>
  )
}
