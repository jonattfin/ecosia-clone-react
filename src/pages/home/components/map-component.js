import { Link } from 'react-router-dom';

import styles from './styles.module.scss';

export default function Component({ counter }) {
  return (
    <section className={styles['map-section']}>
      <div className={styles['wrapper-image']}>
        <div className={styles['above-title']}>Trees planted by ecosia users</div>
        <div className={styles.title__kick}>{counter}</div>
        <div className={styles['facts-grid']}>
          <div className={styles['grid-item']}>
            <div className={styles['grid-item__title']}>15 Million</div>
            <div className={styles['grid-item__text']}>People using Ecosia</div>
          </div>
          <div className={styles['grid-item']}>
            <div className={styles['grid-item__title']}>500+</div>
            <div className={styles['grid-item__text']}>Native species</div>
          </div>
          <div className={styles['grid-item']}>
            <div className={styles['grid-item__title']}>30+</div>
            <div className={styles['grid-item__text']}>Countries</div>
          </div>
          <div className={styles['grid-item']}>
            <div className={styles['grid-item__title']}>60+</div>
            <div className={styles['grid-item__text']}>Active projects</div>
          </div>
        </div>
      </div>
      <div className={styles['report-link']}><Link to="#">Discover our projects...</Link></div>
    </section>
  )
}
