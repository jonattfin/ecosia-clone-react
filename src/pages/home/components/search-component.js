import { FormControl, OutlinedInput } from '@mui/material';
import { Search as SearchIcon } from '@mui/icons-material';

import * as SharedComponents from '../../../shared-components'
import * as Images from './images';

import styles from './styles.module.scss';

export default function Component(props) {
  const { query, setQuery, doSearch, language, counter } = props;

  return (
    <section className={styles['search-section']}>
      <div>
        <img className={styles.logo} src={Images.LogoImage} alt="logo" />
      </div>
      <div className={styles.counter}>
        <FormControl sx={{ m: 1, width: '25ch' }} variant="outlined" className={styles['search-box']}>
          <OutlinedInput
            value={query}
            placeholder="Search the web to plant trees..."
            onChange={(ev) => { setQuery(ev.target.value); }}
            onKeyDown={ev => { if (ev.key === 'Enter') doSearch(query); }}
            endAdornment={<SearchIcon />}
          />
        </FormControl>
        <div>
          {query}
        </div>
        <div className={styles.counter__wrapper}>
          <h1>{language['the-search-engine']}</h1>
          <SharedComponents.Counter counter={counter} />
          <div>&nbsp;</div>
        </div>
      </div>
    </section>
  )
}
