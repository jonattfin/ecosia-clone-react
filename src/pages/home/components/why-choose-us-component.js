import { Link } from 'react-router-dom';

import * as Images from './images';

import styles from './styles.module.scss';

export default function Component() {
  return (
    <section className={styles['why-choose-us-section']}>
      <div className={styles.wrapper}>
        <div className={styles['above-title']}>Why choose Ecosia? </div>
        <div className={styles.title}>We put people and planet before profit</div>
        <div>&nbsp;</div>
        <div className={styles['reasons-grid']}>
          <div className={styles['grid-item']}>
            <div>
              <img src={Images.ProfitsImage} alt="profits" />
            </div>
            <div className={styles['grid-item__title']}>
              We’re a not-for-profit business
            </div>
            <div>
              We dedicate 100% of our profits to climate action, with at least 80% financing tree-planting projects.
            </div>
          </div>
          <div className={styles['grid-item']}>
            <div>
              <img src={Images.PrivacyImage} alt="privacy" />
            </div>
            <div className={styles['grid-item__title']}>
              Ecosia is powered by 200% renewable energy
            </div>
            <div>
              Our solar panels produce twice the amount of energy needed to power all searches with renewables.
            </div>
          </div>
          <div className={styles['grid-item']}>
            <div>
              <img src={Images.WorldImage} alt="world" />
            </div>
            <div className={styles['grid-item__title']}>
              We always put your privacy first
            </div>
            <div>
              We anonymize your searches and don’t create a profile of you. We’re interested in trees, not your data.
            </div>

          </div>
          <div className={styles['grid-item']}>
            <div>
              <img src={Images.CoinImage} alt="coin" />
            </div>
            <div className={styles['grid-item__title']}>
              We are transparent about everything we do
            </div>
            <div>
              We publish detailed financial reports and frequent updates from our tree planting projects.
            </div>
          </div>
        </div>
        <div>
          <div>&nbsp;</div>
          <div className={styles['report-link']}><Link to="#">Other reasons...</Link></div>
        </div>
      </div>
    </section>
  )
}
