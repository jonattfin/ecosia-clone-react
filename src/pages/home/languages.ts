import { Languages } from '../../language';

const languages = {
  [Languages.En]: {
    'search-to-plant-trees': "Search the web to plant trees...",
    'the-search-engine': 'The search engine that plants trees.',
    'where-do-you-plant-trees': 'Where are your trees being planted?',
    'we-plant-30-countries': 'We plant in 30+ countries with local organizations',
    'we-plant-in-brazil': 'Your trees in Brazil protect thousands of endangered plants and animals.',
    'we-plant-in-burkina-faso': 'By planting trees in Burkina Faso, you restore desertified land to its former fertility.',
    'we-plant-in-indonesia': 'In Indonesia, your searches bring back forests on former palm oil plantations while creating alternative sources of income.'
  },
  [Languages.Ro]: {
    'search-to-plant-trees': "(R) Search the web to plant trees...",
    'the-search-engine': '(R) The search engine that plants trees.',
    'where-do-you-plant-trees': '(R) Where are your trees being planted?',
    'we-plant-30-countries': '(R) We plant in 30+ countries with local organizations',
    'we-plant-in-brazil': '(R) Your trees in Brazil protect thousands of endangered plants and animals.',
    'we-plant-in-burkina-faso': '(R) By planting trees in Burkina Faso, you restore desertified land to its former fertility.',
    'we-plant-in-indonesia': '(R) In Indonesia, your searches bring back forests on former palm oil plantations while creating alternative sources of income.'
  },
};

export const getLanguage = (lang: Languages) => {
  return languages[lang];
}
