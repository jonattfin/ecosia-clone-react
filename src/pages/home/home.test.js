import React from 'react';
import renderer from 'react-test-renderer';

import * as Components from './components';
import styles from './styles.module.scss';
import { withRouter } from '../../helpers';

import { getLanguage } from './languages';
import { Languages } from '../../language';

const language = getLanguage(Languages.En)

it('<Components.CountriesComponent/> renders correctly', () => {
  const tree = renderer
    .create(<Components.CountriesComponent {...{ styles, language }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// it('<Components.FinancialComponent/> renders correctly', () => {
//   const tree = renderer
//     .create(<Components.FinancialComponent {...{ styles }} />)
//     .toJSON();
//   expect(tree).toMatchSnapshot();
// });

it('<Components.JoinUsComponent/> renders correctly', () => {
  const tree = renderer
    .create(<Components.JoinUsComponent {...{ styles }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('<Components.MapComponent/> renders correctly', () => {
  const tree = renderer
    .create(withRouter(Components.MapComponent, { styles, language }))
    .toJSON();
  expect(tree).toMatchSnapshot();
});

// it('<Components.PieComponent/> renders correctly', () => {
//   const tree = renderer
//     .create(<Components.PieComponent {...{ styles }} />)
//     .toJSON();
//   expect(tree).toMatchSnapshot();
// });

it('<Components.SearchComponent/> renders correctly', () => {
  const tree = renderer
    .create(<Components.SearchComponent {... { styles, language }} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('<Components.WhyChooseUsComponentw/> renders correctly', () => {
  const tree = renderer
    .create(withRouter(Components.WhyChooseUsComponent, {...{ styles }}))
    .toJSON();
  expect(tree).toMatchSnapshot();
});

