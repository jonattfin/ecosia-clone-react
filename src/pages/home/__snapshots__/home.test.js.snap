// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Components.CountriesComponent/> renders correctly 1`] = `
<section
  className="countries-section"
>
  <div
    className="above-title"
  >
    Where are your trees being planted?
  </div>
  <div
    className="title"
  >
    We plant in 30+ countries with local organizations
  </div>
  <div
    className="countries-grid"
  >
    <div
      className="grid-item"
    >
      <div>
        <img
          alt="brazil"
          className="country"
          src="brazil.webp"
        />
      </div>
      <div
        className="grid-item__title"
      >
        Brazil
      </div>
      <div
        className="grid-item__subtitle"
      >
        Your trees in Brazil protect thousands of endangered plants and animals.
      </div>
    </div>
    <div
      className="grid-item"
    >
      <img
        alt="burkina faso"
        className="country"
        src="burkinafaso.webp"
      />
      <div
        className="grid-item__title"
      >
        Burkina Faso
      </div>
      <div
        className="grid-item__subtitle"
      >
        By planting trees in Burkina Faso, you restore desertified land to its former fertility.
      </div>
    </div>
    <div
      className="grid-item"
    >
      <img
        alt="indonesia"
        className="country"
        src="indonesia.webp"
      />
      <div
        className="grid-item__title"
      >
        Indonesia
      </div>
      <div
        className="grid-item__subtitle"
      >
        In Indonesia, your searches bring back forests on former palm oil plantations while creating alternative sources of income.
      </div>
    </div>
  </div>
</section>
`;

exports[`<Components.JoinUsComponent/> renders correctly 1`] = `
<section
  className="join-section"
>
  <div
    className="main-grid"
  >
    <div
      className="grid-item__title"
    >
      Join 15 million people who already use Ecosia
    </div>
  </div>
</section>
`;

exports[`<Components.MapComponent/> renders correctly 1`] = `
<section
  className="map-section"
>
  <div
    className="wrapper-image"
  >
    <div
      className="above-title"
    >
      Trees planted by ecosia users
    </div>
    <div
      className="title__kick"
    />
    <div
      className="facts-grid"
    >
      <div
        className="grid-item"
      >
        <div
          className="grid-item__title"
        >
          15 Million
        </div>
        <div
          className="grid-item__text"
        >
          People using Ecosia
        </div>
      </div>
      <div
        className="grid-item"
      >
        <div
          className="grid-item__title"
        >
          500+
        </div>
        <div
          className="grid-item__text"
        >
          Native species
        </div>
      </div>
      <div
        className="grid-item"
      >
        <div
          className="grid-item__title"
        >
          30+
        </div>
        <div
          className="grid-item__text"
        >
          Countries
        </div>
      </div>
      <div
        className="grid-item"
      >
        <div
          className="grid-item__title"
        >
          60+
        </div>
        <div
          className="grid-item__text"
        >
          Active projects
        </div>
      </div>
    </div>
  </div>
  <div
    className="report-link"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Discover our projects...
    </a>
  </div>
</section>
`;

exports[`<Components.SearchComponent/> renders correctly 1`] = `
<section
  className="search-section"
>
  <div>
    <img
      alt="logo"
      className="logo"
      src="logo.png"
    />
  </div>
  <div
    className="counter"
  >
    <div
      className="MuiFormControl-root search-box css-r47a1p-MuiFormControl-root"
    >
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="Search the web to plant trees..."
          required={false}
          type="text"
          value=""
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SearchIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
          />
        </svg>
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-hdw1oc"
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="counter__wrapper"
    >
      <h1>
        The search engine that plants trees.
      </h1>
      <div
        className="counter"
      >
        <div
          className="counter__text"
          style={
            Object {
              "fontSize": "4em",
            }
          }
        >
          0
        </div>
        <div>
          Trees planted by Ecosia users
        </div>
      </div>
      <div>
         
      </div>
    </div>
  </div>
</section>
`;

exports[`<Components.WhyChooseUsComponentw/> renders correctly 1`] = `
<section
  className="why-choose-us-section"
>
  <div
    className="wrapper"
  >
    <div
      className="above-title"
    >
      Why choose Ecosia? 
    </div>
    <div
      className="title"
    >
      We put people and planet before profit
    </div>
    <div>
       
    </div>
    <div
      className="reasons-grid"
    >
      <div
        className="grid-item"
      >
        <div>
          <img
            alt="profits"
            src="profits.svg"
          />
        </div>
        <div
          className="grid-item__title"
        >
          We’re a not-for-profit business
        </div>
        <div>
          We dedicate 100% of our profits to climate action, with at least 80% financing tree-planting projects.
        </div>
      </div>
      <div
        className="grid-item"
      >
        <div>
          <img
            alt="privacy"
            src="privacy.svg"
          />
        </div>
        <div
          className="grid-item__title"
        >
          Ecosia is powered by 200% renewable energy
        </div>
        <div>
          Our solar panels produce twice the amount of energy needed to power all searches with renewables.
        </div>
      </div>
      <div
        className="grid-item"
      >
        <div>
          <img
            alt="world"
            src="world.svg"
          />
        </div>
        <div
          className="grid-item__title"
        >
          We always put your privacy first
        </div>
        <div>
          We anonymize your searches and don’t create a profile of you. We’re interested in trees, not your data.
        </div>
      </div>
      <div
        className="grid-item"
      >
        <div>
          <img
            alt="coin"
            src="coin.svg"
          />
        </div>
        <div
          className="grid-item__title"
        >
          We are transparent about everything we do
        </div>
        <div>
          We publish detailed financial reports and frequent updates from our tree planting projects.
        </div>
      </div>
    </div>
    <div>
      <div>
         
      </div>
      <div
        className="report-link"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Other reasons...
        </a>
      </div>
    </div>
  </div>
</section>
`;
